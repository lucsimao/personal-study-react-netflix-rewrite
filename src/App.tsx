import React, { useEffect, useState } from 'react';
import { getCategories, getMovieWithVideo } from './services/movies-service';

import Body from './components/body/body';
import { Category } from './domain/models/category';
import { GlobalStyles } from './GlobalStyles';
import Header from './components/header/header';
import { HttpClient } from './infra/interfaces/http-client';
import { Movie } from './domain/models/movie';

const movie: Movie = {
  id: '585245',
  title: 'Clifford: O Gigante Cão Vermelho',
  overview: `Baseado na adorada história dos livros escolares escritos por Norman
        Bridwell, Clifford irá mostrar ao mundo como é possível amar em grande!`,
  imagePath: 'g51VeVAIM7q9ZqpZRlPEqqhLThR.jpg',
  videoUrl: 'xlGOCCsValCs',
};

const App: React.FC<{ httpClient: HttpClient }> = (props) => {
  const [categories, setCategories] = useState<Category[]>([]);
  const [mainMovie, setMainMovie] = useState<Movie>(movie);

  const updateCategories = async () => {
    const result = await getCategories(props.httpClient);
    setCategories(result);
  };

  const updateMovie = async () => {
    const result = await getMovieWithVideo(props.httpClient);
    setMainMovie(result);
  };

  useEffect(() => {
    updateCategories();
    updateMovie();
  }, []);

  return (
    <>
      <GlobalStyles />
      <Header />
      <Body categories={categories} mainMovie={mainMovie} />
    </>
  );
};

export default App;
