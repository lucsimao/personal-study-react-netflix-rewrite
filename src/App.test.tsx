import 'jest-styled-components';

import App from './App';
import { HttpClient } from './infra/interfaces/http-client';
import { render } from '@testing-library/react';

const makeHttpClientStub = (): jest.Mocked<HttpClient> => ({
  get: jest.fn().mockResolvedValue({ results: [] }),
});

test('renders learn react link', async () => {
  const httpClientStub = makeHttpClientStub();
  const app = await render(<App httpClient={httpClientStub} />);

  expect(app).toMatchSnapshot();
});
