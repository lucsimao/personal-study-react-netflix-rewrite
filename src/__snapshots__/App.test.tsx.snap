// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: absolute;
  top: 85vh;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  height: 100vh;
  width: 40vw;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 3.1vw;
  padding-bottom: 4vh;
}

.c14 {
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.5vh;
  padding-left: 2.6em;
  padding-right: 2.6em;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0px;
  border-radius: 4px;
  cursor: pointer;
  overflow: visible;
}

.c14:hover {
  background-color: #aaaaaa;
}

.c14 span {
  margin-left: 14px;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 1.7em;
}

.c13 {
  margin-bottom: 16vw;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c15 {
  margin-left: 15px;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.6rem;
  padding-left: 1.8rem;
  padding-right: 1.8rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0px;
  border-radius: 4px;
  cursor: pointer;
  overflow: visible;
  background-color: rgba(109,109,110,0.7);
  color: white;
}

.c15:hover {
  background-color: rgba(49,49,50,0.7);
}

.c15 span {
  margin-left: 14px;
  white-space: nowrap;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 1.76em;
}

.c12 {
  color: #fff;
  font-weight: 400;
  line-height: normal;
  width: 100%;
  font-size: 1.4vw;
  text-shadow: 2px 2px 4px rgb(0 0 0 / 45%);
  margin-top: 0.1vw;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  margin-bottom: 25px;
}

.c11 {
  color: #fff;
  font-weight: 400;
  line-height: normal;
  width: 100%;
  font-size: 3.5vw;
  text-shadow: 4px 4px 8px rgb(0 0 0);
  margin-top: 0.1vw;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  margin-bottom: 25px;
}

.c16 {
  position: absolute;
  top: 0;
  overflow: hidden;
  z-index: -1;
}

.c16:before {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  content: '';
  background-image: linear-gradient( to bottom, rgba(20,20,20,0.2) 0, rgba(20,20,20,0.2) 50%, rgba(20,20,20,0.6) 80%, rgba(20,20,20,0.9) 90%, #141414 95%, #141414 100% );
}

.c17 {
  width: 100vw;
  height: 56.25vw;
  overflow: hidden;
  pointer-events: none;
  top: 0px;
  z-index: -1;
  margin-bottom: -10vh;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: black;
  height: 68px;
  padding: 0px 60px 0px 60px;
}

.c1 {
  cursor: pointer;
  content: url('https://upload.wikimedia.org/wikipedia/commons/0/08/Netflix_2015_logo.svg');
  height: 1.563em;
  margin-right: 25px;
}

.c3 {
  list-style-type: none;
  margin-left: 20px;
  font-size: 0.88em;
  white-space: nowrap;
}

.c4 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 300;
}

.c5 {
  font-weight: 450;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-top: 1em;
  margin-bottom: 1em;
  color: white;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: default;
}

.c6 {
  position: absolute;
  right: 60px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
}

.c7 {
  cursor: pointer;
  content: url('https://occ-0-1768-3852.1.nflxso.net/dnm/api/v6/K6hjPJd6cR6FpVELC5Pd6ovHRSk/AAAABbme8JMz4rEKFJhtzpOKWFJ_6qX-0y5wwWyYvBhWS0VKFLa289dZ5zvRBggmFVWVPL2AAYE8xevD4jjLZjWumNo.png?r=a41');
  height: 2em;
  margin-left: 20px;
  border-radius: 4px;
}

.c8 {
  fill: 'white';
  margin-left: 10px;
  -webkit-transform: rotate( 0deg );
  -ms-transform: rotate( 0deg );
  transform: rotate( 0deg );
  -webkit-transition: 367ms;
  transition: 367ms;
}

<body>
    <div>
      <div
        class="c0"
      >
        <a
          class="c1"
          data-testid="netflix-logo"
          href="/"
        />
        <ul
          class="c2"
          data-testid="navigation-items"
        >
          <li
            class="c3"
          >
            <a
              class="c4 c5"
              href="/browse"
            >
              Início
            </a>
          </li>
          <li
            class="c3"
          >
            <a
              class="c4"
              href="/browse/genre/83"
            >
              Séries
            </a>
          </li>
          <li
            class="c3"
          >
            <a
              class="c4"
              href="/browse/genre/34399"
            >
              Filmes
            </a>
          </li>
          <li
            class="c3"
          >
            <a
              class="c4"
              href="/latest"
            >
              Bombando
            </a>
          </li>
          <li
            class="c3"
          >
            <a
              class="c4"
              href="/browse/my-list"
            >
              Minha lista
            </a>
          </li>
        </ul>
        <div
          class="c6"
        >
          <ul
            class="c2"
            data-testid="infantil-navigation"
          >
            <svg
              fill="currentColor"
              height="19"
              stroke="currentColor"
              stroke-width="0"
              style="fill: white; margin-left: 20px;"
              viewBox="0 0 512 512"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
            <li
              class="c3"
            >
              <a
                class="c4"
                href="/Kids"
              >
                Infantil
              </a>
            </li>
            <svg
              fill="currentColor"
              height="29"
              stroke="currentColor"
              stroke-width="0"
              style="stroke-width: 10; fill: white; margin-left: 20px;"
              viewBox="0 0 512 512"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 464c22.779 0 41.411-18.719 41.411-41.6h-82.823c0 22.881 18.633 41.6 41.412 41.6zm134.589-124.8V224.8c0-63.44-44.516-117.518-103.53-131.041V79.2c0-17.682-13.457-31.2-31.059-31.2s-31.059 13.518-31.059 31.2v14.559c-59.015 13.523-103.53 67.601-103.53 131.041v114.4L80 380.8v20.8h352v-20.8l-41.411-41.6z"
              />
            </svg>
            <a
              class="c7"
            />
            <svg
              class="c8"
              fill="currentColor"
              height="15"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              />
            </svg>
          </ul>
        </div>
      </div>
      <div>
        <div
          class="c9"
        >
          <div
            class="c10"
          >
            <div
              class="c11"
            >
              Clifford: O Gigante Cão Vermelho
            </div>
            <div
              class="c12"
            >
              Baseado na adorada história dos livros escolares escritos por Norman
        Bridwell, Clifford irá mostrar ao mundo como é possível amar em grande!
            </div>
            <div
              class="c13"
            >
              <button
                class="c14"
              >
                <svg
                  fill="currentColor"
                  height="2.2em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="2.2em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
                  />
                </svg>
                <span>
                  Assistir
                </span>
              </button>
              <button
                class="c15"
              >
                <svg
                  fill="none"
                  height="2.8em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="2.8em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 10.9794C11 10.4271 11.4477 9.97937 12 9.97937C12.5523 9.97937 13 10.4271 13 10.9794V16.9794C13 17.5317 12.5523 17.9794 12 17.9794C11.4477 17.9794 11 17.5317 11 16.9794V10.9794Z"
                    fill="currentColor"
                  />
                  <path
                    d="M12 6.05115C11.4477 6.05115 11 6.49886 11 7.05115C11 7.60343 11.4477 8.05115 12 8.05115C12.5523 8.05115 13 7.60343 13 7.05115C13 6.49886 12.5523 6.05115 12 6.05115Z"
                    fill="currentColor"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2ZM4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
                <span>
                  Mais informações
                </span>
              </button>
            </div>
          </div>
          <div
            class="c16"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              class="c17"
              frameborder="0"
              src="https://www.youtube.com/embed/xlGOCCValCs?version=3&rel=0&autoplay=1&controls=0&&showinfo=0&loop=1&mute=1&start=50&showinfo=0&color=white&iv_load_policy=3&playlist=xlGOCCValCs"
              title="YouTube video player"
            />
          </div>
        </div>
        <ul
          class="c18"
        />
      </div>
    </div>
  </body>,
  "container": .c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: absolute;
  top: 85vh;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  height: 100vh;
  width: 40vw;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 3.1vw;
  padding-bottom: 4vh;
}

.c14 {
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.5vh;
  padding-left: 2.6em;
  padding-right: 2.6em;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0px;
  border-radius: 4px;
  cursor: pointer;
  overflow: visible;
}

.c14:hover {
  background-color: #aaaaaa;
}

.c14 span {
  margin-left: 14px;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 1.7em;
}

.c13 {
  margin-bottom: 16vw;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c15 {
  margin-left: 15px;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.6rem;
  padding-left: 1.8rem;
  padding-right: 1.8rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0px;
  border-radius: 4px;
  cursor: pointer;
  overflow: visible;
  background-color: rgba(109,109,110,0.7);
  color: white;
}

.c15:hover {
  background-color: rgba(49,49,50,0.7);
}

.c15 span {
  margin-left: 14px;
  white-space: nowrap;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 1.76em;
}

.c12 {
  color: #fff;
  font-weight: 400;
  line-height: normal;
  width: 100%;
  font-size: 1.4vw;
  text-shadow: 2px 2px 4px rgb(0 0 0 / 45%);
  margin-top: 0.1vw;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  margin-bottom: 25px;
}

.c11 {
  color: #fff;
  font-weight: 400;
  line-height: normal;
  width: 100%;
  font-size: 3.5vw;
  text-shadow: 4px 4px 8px rgb(0 0 0);
  margin-top: 0.1vw;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  margin-bottom: 25px;
}

.c16 {
  position: absolute;
  top: 0;
  overflow: hidden;
  z-index: -1;
}

.c16:before {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  content: '';
  background-image: linear-gradient( to bottom, rgba(20,20,20,0.2) 0, rgba(20,20,20,0.2) 50%, rgba(20,20,20,0.6) 80%, rgba(20,20,20,0.9) 90%, #141414 95%, #141414 100% );
}

.c17 {
  width: 100vw;
  height: 56.25vw;
  overflow: hidden;
  pointer-events: none;
  top: 0px;
  z-index: -1;
  margin-bottom: -10vh;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: black;
  height: 68px;
  padding: 0px 60px 0px 60px;
}

.c1 {
  cursor: pointer;
  content: url('https://upload.wikimedia.org/wikipedia/commons/0/08/Netflix_2015_logo.svg');
  height: 1.563em;
  margin-right: 25px;
}

.c3 {
  list-style-type: none;
  margin-left: 20px;
  font-size: 0.88em;
  white-space: nowrap;
}

.c4 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Netflix Sans','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 300;
}

.c5 {
  font-weight: 450;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-top: 1em;
  margin-bottom: 1em;
  color: white;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: default;
}

.c6 {
  position: absolute;
  right: 60px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
}

.c7 {
  cursor: pointer;
  content: url('https://occ-0-1768-3852.1.nflxso.net/dnm/api/v6/K6hjPJd6cR6FpVELC5Pd6ovHRSk/AAAABbme8JMz4rEKFJhtzpOKWFJ_6qX-0y5wwWyYvBhWS0VKFLa289dZ5zvRBggmFVWVPL2AAYE8xevD4jjLZjWumNo.png?r=a41');
  height: 2em;
  margin-left: 20px;
  border-radius: 4px;
}

.c8 {
  fill: 'white';
  margin-left: 10px;
  -webkit-transform: rotate( 0deg );
  -ms-transform: rotate( 0deg );
  transform: rotate( 0deg );
  -webkit-transition: 367ms;
  transition: 367ms;
}

<div>
    <div
      class="c0"
    >
      <a
        class="c1"
        data-testid="netflix-logo"
        href="/"
      />
      <ul
        class="c2"
        data-testid="navigation-items"
      >
        <li
          class="c3"
        >
          <a
            class="c4 c5"
            href="/browse"
          >
            Início
          </a>
        </li>
        <li
          class="c3"
        >
          <a
            class="c4"
            href="/browse/genre/83"
          >
            Séries
          </a>
        </li>
        <li
          class="c3"
        >
          <a
            class="c4"
            href="/browse/genre/34399"
          >
            Filmes
          </a>
        </li>
        <li
          class="c3"
        >
          <a
            class="c4"
            href="/latest"
          >
            Bombando
          </a>
        </li>
        <li
          class="c3"
        >
          <a
            class="c4"
            href="/browse/my-list"
          >
            Minha lista
          </a>
        </li>
      </ul>
      <div
        class="c6"
      >
        <ul
          class="c2"
          data-testid="infantil-navigation"
        >
          <svg
            fill="currentColor"
            height="19"
            stroke="currentColor"
            stroke-width="0"
            style="fill: white; margin-left: 20px;"
            viewBox="0 0 512 512"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <li
            class="c3"
          >
            <a
              class="c4"
              href="/Kids"
            >
              Infantil
            </a>
          </li>
          <svg
            fill="currentColor"
            height="29"
            stroke="currentColor"
            stroke-width="0"
            style="stroke-width: 10; fill: white; margin-left: 20px;"
            viewBox="0 0 512 512"
            width="29"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 464c22.779 0 41.411-18.719 41.411-41.6h-82.823c0 22.881 18.633 41.6 41.412 41.6zm134.589-124.8V224.8c0-63.44-44.516-117.518-103.53-131.041V79.2c0-17.682-13.457-31.2-31.059-31.2s-31.059 13.518-31.059 31.2v14.559c-59.015 13.523-103.53 67.601-103.53 131.041v114.4L80 380.8v20.8h352v-20.8l-41.411-41.6z"
            />
          </svg>
          <a
            class="c7"
          />
          <svg
            class="c8"
            fill="currentColor"
            height="15"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            />
          </svg>
        </ul>
      </div>
    </div>
    <div>
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <div
            class="c11"
          >
            Clifford: O Gigante Cão Vermelho
          </div>
          <div
            class="c12"
          >
            Baseado na adorada história dos livros escolares escritos por Norman
        Bridwell, Clifford irá mostrar ao mundo como é possível amar em grande!
          </div>
          <div
            class="c13"
          >
            <button
              class="c14"
            >
              <svg
                fill="currentColor"
                height="2.2em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="2.2em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
                />
              </svg>
              <span>
                Assistir
              </span>
            </button>
            <button
              class="c15"
            >
              <svg
                fill="none"
                height="2.8em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="2.8em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 10.9794C11 10.4271 11.4477 9.97937 12 9.97937C12.5523 9.97937 13 10.4271 13 10.9794V16.9794C13 17.5317 12.5523 17.9794 12 17.9794C11.4477 17.9794 11 17.5317 11 16.9794V10.9794Z"
                  fill="currentColor"
                />
                <path
                  d="M12 6.05115C11.4477 6.05115 11 6.49886 11 7.05115C11 7.60343 11.4477 8.05115 12 8.05115C12.5523 8.05115 13 7.60343 13 7.05115C13 6.49886 12.5523 6.05115 12 6.05115Z"
                  fill="currentColor"
                />
                <path
                  clip-rule="evenodd"
                  d="M12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2ZM4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span>
                Mais informações
              </span>
            </button>
          </div>
        </div>
        <div
          class="c16"
        >
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            class="c17"
            frameborder="0"
            src="https://www.youtube.com/embed/xlGOCCValCs?version=3&rel=0&autoplay=1&controls=0&&showinfo=0&loop=1&mute=1&start=50&showinfo=0&color=white&iv_load_policy=3&playlist=xlGOCCValCs"
            title="YouTube video player"
          />
        </div>
      </div>
      <ul
        class="c18"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
